scripts/ci/pr-bot/processPrUpdate.ts

```diff
@@ -0,0 +1,198 @@
+const { PATH_TO_CONFIG_FILE } = require("./shared/constants");
+
+async function areReviewersAssigned(
+  pullNumber: number,
+  stateClient: typeof PersistentState
+): Promise<boolean> {
+  const prState = await stateClient.getPrState(pullNumber);
+  return Object.values(prState.reviewersAssignedForLabels).length > 0;
+}
+
+async function processPrComment(
+  payload,
+  stateClient: typeof PersistentState,
+  reviewerConfig: typeof ReviewerConfig
+) {
+  const commentContents = payload.comment.body;
+  const commentAuthor = payload.sender.login;
+  console.log(commentContents);
+  if (
+    await processCommand(
+      payload,
+      commentAuthor,
+      commentContents,
+      stateClient,
+      reviewerConfig
+    )
+  ) {
+    // If we've processed a command, don't worry about trying to change the attention set.
+    // This is not a meaningful push or comment from the author.
+    console.log("Processed command");
+    return;
+  }
+
+  // If comment was from the author, we should shift attention back to the reviewers.
+  console.log(
+    "No command to be processed, checking if we should shift attention to reviewers"
+  );
+  const pullAuthor =
+    payload.issue?.user?.login || payload.pull_request?.user?.login;
+  if (pullAuthor == commentAuthor) {
+    await setNextActionReviewers(payload, stateClient);
+  } else {
+    console.log(
+      `Comment was from ${commentAuthor}, not author: ${pullAuthor}. No action to take.`
+    );
+  }
+}
+
+// On approval from a reviewer we have assigned, assign committer if one not already assigned
+async function processPrReview(
+  payload,
+  stateClient: typeof PersistentState,
+  reviewerConfig: typeof ReviewerConfig
+) {
+  if (payload.review.state != "approved") {
+    return;
+  }
+
+  const pullNumber = payload.issue?.number || payload.pull_request?.number;
+  if (!(await areReviewersAssigned(pullNumber, stateClient))) {
+    return;
+  }
+
+  let prState = await stateClient.getPrState(pullNumber);
+  // TODO(damccorm) - also check if the author is a committer, if they are don't auto-assign a committer
+  if (await prState.isAnyAssignedReviewerCommitter()) {
+    return;
+  }
+
+  let labelOfReviewer = prState.getLabelForReviewer(payload.sender.login);
+  if (labelOfReviewer) {
+    let reviewersState = await stateClient.getReviewersForLabelState(
+      labelOfReviewer
+    );
+    let availableReviewers =
+      reviewerConfig.getReviewersForLabel(labelOfReviewer);
+    let chosenCommitter = await reviewersState.assignNextCommitter(
+      availableReviewers
+    );
+    prState.reviewersAssignedForLabels[labelOfReviewer] = chosenCommitter;
+
+    // Set next action to committer
+    await addPrComment(
+      pullNumber,
+      commentStrings.assignCommitter(chosenCommitter)
+    );
+    const existingLabels =
+      payload.issue?.labels || payload.pull_request?.labels;
+    await nextActionReviewers(pullNumber, existingLabels);
+    prState.nextAction = "Reviewers";
```

Comment:
`"Reviewers"` is used multiple times. Please add a constant string at the top of the file.